src/main/java/org/generationcp/middleware/api/germplasmlist/GermplasmListServiceImpl.java

```diff
@@ -81,6 +150,116 @@ public GermplasmListGeneratorDTO create(final GermplasmListGeneratorDTO request,
 		return request;
 	}
 
+	@Override
+	public List<GermplasmListData> addGermplasmListData(final List<GermplasmListData> data) {
+
+		final List<GermplasmListData> idGermplasmListDataSaved = new ArrayList<>();
+		try {
+			final List<Integer> deletedListEntryIds = new ArrayList<>();
+			data.forEach(germplasmListData -> {
+				String groupName = germplasmListData.getGroupName();
+				if (groupName.length() > MAX_CROSS_NAME_SIZE) {
+					groupName = groupName.substring(0, MAX_CROSS_NAME_SIZE - 1);
+					groupName = groupName + TRUNCATED;
+					germplasmListData.setGroupName(groupName);
+				}
+
+				final GermplasmListData recordSaved = this.daoFactory.getGermplasmListDataDAO().saveOrUpdate(germplasmListData);
```

Comment:
do we need recordSaved? I think hibernate modify the object and set the id, so we could use germplasmListData instead @lik-83